src/controllers/auth.js

```diff
@@ -0,0 +1,132 @@
+const jwt = require('jsonwebtoken');
+//const bcrypt = require('bcrypt');
+const bcrypt = require('bcryptjs');
+const User = require('../models/users');
+const { generateToken } = require('../middleware/checkAuth');
+require('dotenv').config();
+const { validationResult } = require('express-validator');
+const {
+  validateUsername,
+  validateEmail,
+  validatePassword,
+  handleValidationErrors,
+} = require('../middleware/validation');
+
+
+const signUp = async (req, res) => {
+  console.log('Request Body:', req.body);
+
+  try {
+    // Validate the request body using the validation functions
+    await Promise.all([
```

Comment:
what is exactly the reason for this mongoose has its own validation for what is required